### PR TITLE
Implement, test, doc git descriptions for things

### PR DESCRIPTION
**Pull request by kshyatt**
Now you can ask for detailed descriptions of a repo's workdir or a commit-like object within it. cc @staticfloat.

### CODE REVIEW

File: base/libgit2/repository.jl
Comment (ararslan) [inline base/libgit2/repository.jl]: I don't think there should be a hyphen between `git` and `describe`; `git describe` is the CLI AFAICT

**Comment by ararslan** [comment]
It might be nice to have the description interface be simply `LibGit2.describe(commitish, ...)`. That's a little more clear IMO than having to directly construct a `GitDescribeResult`.

**Comment by kshyatt** [comment]
I did it this way because it's a type and otherwise it lacks a constructor.

**Comment by ararslan** [comment]
`describe` could just be an alias for the constructor

File: test/libgit2.jl
Comment (tkelman) [inline test/libgit2.jl]: don't add empty lines as the first or last thing in a block

**Comment by tkelman** [comment]
we're in feature freeze, not the best time to be wrapping new api's

**Comment by kshyatt** [comment]
@staticfloat asked for it, we can wait until we're out of FF to merge if you want.

**Comment by kshyatt** [comment]
@tkelman I've rebased this against master. We're off feature freeze, right?

**Comment by kshyatt** [comment]
And bumping again. Can someone ( @ararslan ? @simonbyrne ? ) please review this?

**Review by simonbyrne — APPROVED** [review]
I made the small doc change suggested by @ararslan, LGTM once CI is done.

File: base/libgit2/repository.jl
Comment (tkelman) [inline base/libgit2/repository.jl]: does the same apply here?

**Pull request merged**
